### ISSUE TITLE
Using Emacs with encryption hangs terminal and emacs processes

### ISSUE THREAD

**Issue by atfornes** [keybase/keybase-issues#2485]
I was trying [EasyPG](https://www.emacswiki.org/emacs/EasyPG)  with emacs to encrypt a not yet existing file inside my private keybase folder. When saving that file from emacs (which should have saved it encrypted), both the emacs process and  the terminal from which it was started hanged so badly they could not even be killed by `kill -9` command, with  `task mysqld:xxx blocked for more than 120 seconds` error messages. See [this stackexchange thread](http://unix.stackexchange.com/questions/181505/what-does-task-mysqldxxx-blocked-for-more-than-120-seconds-mean) for details about this failure.

I have try to send a log with `keybase log send` but an error message appears saying  `"Error API network error: ... (Client.Timeout exceeded while awaiting headers)" `

**Comment by strib** [comment]
Hi @atfornes.  Hmm, that's trouble, sorry for the inconvenience.  Do you know about what day/time this happened?  There's a lot going on, including a period of time today where it looks like not all the processes were running -- kbfs was running, but it couldn't connect to the local `keybase` daemon that manages your identity.  Or if you could send me the filename you were working on (if it's secret, you could use `keybase encrypt strib@github` and post the output here), that might help me pin down when the problem happened.

Does KBFS work most of the time for you, and it just froze this once?

**Comment by atfornes** [comment]
Hi @strib, I was working on the file `meta.org` and the problem happened when trying to save it as `meta.org.gpg` with some metadata in the file which would make emacs to encrypt it to my public key.

Date/Time was approximatelly 10 to 20 minutes before this issue report (~ August 24th, 10:45 GTM +2).

I have used keybase occasionally, but I don't use it on a daily basis. Some months ago I had [another issue](https://github.com/keybase/keybase-issues/issues/2239) with keybase also using emacs which was reported and fixed: 

Thanks for the quick reply.

**Comment by strib** [comment]
Thanks for the info.  It looks to me like you may have hit a deadlock in `favorites.go` that was fixed in a later build of KBFS (via keybase/kbfs#178).

Specifically, it seems like emacs might have hung on this request:

```
2016-08-14T22:23:34.410052 ▶ [DEBU kbfs(kbfsfuse) folderlist.go:100] 1bac FL Lookup atenorio [tags:FID=_wmALm9aXOIGRliksQHyWQ]
2016-08-14T22:23:34.411014 ▶ [DEBU kbfs(kbfsfuse) folderlist.go:89] 1bad adding atenorio to favorites [tags:FID=_wmALm9aXOIGRliksQHyWQ]
```

So hopefully upgrading to the latest build will prevent this issue from happening again!

**Comment by atfornes** [comment]
My keybase version is 1.0.16-20160712160059+3c54cc3

Is the fix coming in an automatic update or should I update it manually? Since the update of the version I have shared, Keybase has stopped asking me to download new versions as it used to do.

**Comment by strib** [comment]
I don't think there is an automatic updater on Linux, but you're right that there used to be something that notified you of available updates.  Not sure what happened to that (cc @cjb @gabriel?).

For now please update manually via `apt-get`.

**Comment by cjb** [comment]
Yeah, that dialog was super annoying.  We're just depending on your package manager for the updates from now on -- no automatic updates from us.

**Comment by atfornes** [comment]
I have now the last version available through apt-get: 1.0.17-20160815160038.99c5a56, I will inform if the issue happens again.

Thanks for the quick response on this issue, it is really appreciated.

PS: @cjb, I agree, the dialog was annoying. But still... it would be nice to know that there are new versions available, maybe from the keybase GUI or the login dialog. Anyway that is another issue.
